sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/CosmosDiagnosticsTest.java

```diff
@@ -477,6 +487,80 @@ public void serializationOnVariousScenarios() {
+        try {
+            client = new CosmosClientBuilder()
+                .endpoint(TestConfigurations.HOST)
+                .key(TestConfigurations.MASTER_KEY)
+                .contentResponseOnWriteEnabled(true)
+                .directMode()
+                .buildClient();
+            CosmosContainer container =
+                client.getDatabase(cosmosAsyncContainer.getDatabase().getId()).getContainer(cosmosAsyncContainer.getId());
+            InternalObjectNode internalObjectNode = getInternalObjectNode();
+            CosmosItemResponse<InternalObjectNode> writeResourceResponse = container.createItem(internalObjectNode);
+            //Success address resolution client side statistics
+            assertThat(writeResourceResponse.getDiagnostics().toString()).contains("addressResolutionStatistics");
+            assertThat(writeResourceResponse.getDiagnostics().toString()).contains("\"inflightRequest\":false");
+            assertThat(writeResourceResponse.getDiagnostics().toString()).doesNotContain("endTime=\"null\"");
+            assertThat(writeResourceResponse.getDiagnostics().toString()).contains("\"errorMessage\":null");
+            assertThat(writeResourceResponse.getDiagnostics().toString()).doesNotContain("\"errorMessage\":\"io.netty" +
+                ".channel.AbstractChannel$AnnotatedConnectException: Connection refused: no further information");
+
+            client.close();
+            client = new CosmosClientBuilder()
+                .endpoint(TestConfigurations.HOST)
+                .key(TestConfigurations.MASTER_KEY)
+                .contentResponseOnWriteEnabled(true)
+                .directMode()
+                .buildClient();
+            container =
+                client.getDatabase(cosmosAsyncContainer.getDatabase().getId()).getContainer(cosmosAsyncContainer.getId());
+            AsyncDocumentClient asyncDocumentClient = client.asyncClient().getContextClient();
+            GlobalAddressResolver addressResolver = (GlobalAddressResolver) FieldUtils.readField(asyncDocumentClient,
+                "addressResolver", true);
+
+            @SuppressWarnings("rawtypes")
+            Map addressCacheByEndpoint = (Map) FieldUtils.readField(addressResolver,
+                "addressCacheByEndpoint",
+                true);
+            Object endpointCache = addressCacheByEndpoint.values().toArray()[0];
+            GatewayAddressCache addressCache = (GatewayAddressCache) FieldUtils.readField(endpointCache, "addressCache", true);
+
+            HttpClient httpClient = httpClient(true);
+            FieldUtils.writeField(addressCache, "httpClient", httpClient, true);
+            new Thread(() -> {
+                try {
+                    Thread.sleep(5000);
+                    HttpClient httpClient1 = httpClient(false);
+                    FieldUtils.writeField(addressCache, "httpClient", httpClient1, true);
+                } catch (Exception e) {
+                    fail(e.getMessage());
```

Comment:
I suspect this failure/assertion doesn't have any effect as this is happening on a different thread which is not joined.